src/main/java/org/craftercms/studio/api/v2/content/LifeCycleContentProvider.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2007-2025 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.craftercms.studio.api.v2.content;
+
+import org.springframework.util.function.ThrowingSupplier;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+
+import static org.apache.commons.io.FileUtils.deleteQuietly;
+import static org.craftercms.studio.api.v2.utils.StudioUtils.createTempFile;
+
+/**
+ * The {@link LifeCycleContentProvider} presents a facade for retrieving
+ * content for content life cycle operations. It can be used to provide content
+ * either from a file path or an InputStream.
+ */
+public class LifeCycleContentProvider implements AutoCloseable {
+	private final ThrowingSupplier<Path> pathSupplier;
+	private Path filePath;
+
+	private LifeCycleContentProvider(ThrowingSupplier<Path> pathSupplier) {
+		this.pathSupplier = pathSupplier;
+	}
+
+	/**
+	 * Gets the content as an InputStream.
+	 *
+	 * @return an InputStream for the content
+	 * @throws IOException if there is any error obtaining the content
+	 */
+	public InputStream getContent() throws IOException {
+		return new FileInputStream(filePath().toFile());
```

Comment:
Who is closing this? See the rabbit's comments.

Reply:
I will add the suggested comment to the method